react-components/src/utilities/instanceIds/toKey.ts

```diff
@@ -3,15 +3,15 @@ import { assertNever } from '../assertNever';
 import { isDmsInstance, isExternalId, isInternalId } from './typeGuards';
 import { type InstanceReference } from './types';
 
-export type InstanceReferenceKey = string;
+export type InstanceReferenceKey = string | number;
```

Comment:
The reason for this change is that it simplifies the interaction between `InstanceIdKey` and `InstanceReferenceKey` (`InstanceIdKey` could be a `number` earlier, and allowing `InstanceReferenceKey` to be that as well make them easier to interchange.